lighthouse-core/scripts/compare-runs.js

```diff
@@ -205,14 +213,24 @@ function aggregateResults(name) {
   for (const lhrPath of glob.sync(`${outputDir}/**/lhr-*.json`)) {
     const lhrJson = fs.readFileSync(lhrPath, 'utf-8');
     /** @type {LH.Result} */
-    const lhr = JSON.parse(lhrJson);
+    const lhr = lhrJson && JSON.parse(lhrJson);
+    if (!lhr || !lhr.audits) {
+      console.warn(`could not parse ${lhrPath}`);
+      continue;
+    }
 
-    const metrics = lhr.audits.metrics ?
+    if (argv.urlFilter && !lhr.requestedUrl.includes(argv.urlFilter)) continue;
+
+    const metrics = lhr.audits.metrics && lhr.audits.metrics.details ?
     /** @type {!LH.Audit.Details.Table} */ (lhr.audits.metrics.details).items[0] :
       {};
     const allEntries = {
       metric: Object.entries(metrics).filter(([name]) => !name.endsWith('Ts')),
       timing: lhr.timing.entries.map(entry => ([entry.name, entry.duration])),
+      categories: Object.values(lhr.categories).reduce((acc, cur) => {
+        acc.push([cur.id, (cur.score || 0) * 100]);
+        return acc;
+      }, /** @type {Array<[string, number]>} */ ([])),
```

Comment:
Seems weird to output this as an array of `strings` and `numbers`? And not as a map of `strings`:`numbers`? Any reason to output like this?

Reply:
The `allEntries` object expects this format. Did you see how it's used in the following loop, L241? 

The format comes naturally from `Object.entries(metrics) -> Array<[kind: string, value: number]>`